packages/next/client/image.tsx

```diff
@@ -107,15 +108,26 @@ export type ImageProps = Omit<
   objectFit?: ImgElementStyle['objectFit']
   objectPosition?: ImgElementStyle['objectPosition']
   onLoadingComplete?: OnLoadingComplete
-}
+} & (
+    | {
+        layout?: Omit<LayoutValue, 'raw'>
+      }
+    | {
+        layout: 'raw'
+        style: ImgElementStyle
+      }
+  )
 
 function getWidths(
   { deviceSizes, allSizes }: ImageConfig,
   width: number | undefined,
   layout: LayoutValue,
   sizes: string | undefined
 ): { widths: number[]; kind: 'w' | 'x' } {
-  if (sizes && (layout === 'fill' || layout === 'responsive')) {
+  if (
```

Comment:
I'm including `raw` with the `w` type layouts because I think the main use case for raw images are going to be places where people are manually resizing the images using styling, and therefore we'll want to have a full `srcset`.

I considered trying to inspect the properties of `style` if it exists, to make optimizations to the `srcset`, but I think that would probably be more complexity than it's worth.

Reply:
Would it help if we had two new layouts: `raw-responsive` and `raw-fixed`?

Reply:
Maybe we can see if there's demand for that first? I think we should try to avoid adding two new layout modes if we can, to keep it simple

Reply:
"simple" as in engineering overhead or simple as in too-many-choices-for-the-user?

Because I think the engineering overhead here would be small to add support for both.

Also, I feel like if I saw `layout=raw` as a user, I would assume it mimics a raw `<img>` as close as possible which would be closest to `layout=fixed` rather than `layout=responsive`.

Reply:
Simple as in too-many-choices-for-the-user (sorry - my comment was pretty vague). I think we have to be careful about adding a layout mode for each expectation a user could have because each mode adds mental load for new users. I think it's fine if we make a choice here and document it clearly. If we find that people's expectations are different than we expected, we can always add more modes (harder to take them away)

Reply:
I think I found the right compromise here. The "raw" layout behaves like the `fixed` layout (srcset limited to exactly 1x and 2x the width) when there is no `sizes` prop present. It behaves like `fill` (large srcset) if `sizes` is provided.

I think the missing element here is going to be providing a warning via performanceObserver if the image is rendered much smaller than intrinsic. This will allow us to alert people if they ought to be using `sizes` on their `layout="raw"` image.